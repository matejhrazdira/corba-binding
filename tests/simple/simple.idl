module IncludedIdl {
  struct IncludedStruct
  {
    string someMember;
  };
};

module SimpleIdl
{

  const long IDL_VERSION = 1;
  const long COMPOSED_CONST = 2 + 6 * (1 - 3) * ( -5/2) - 5 + ::SimpleIdl::IDL_VERSION;
  const string STRING_CONSTANT = "This is some string!";

  typedef long long time;

  typedef long long someReally, someReallyWeird, someReallyWeirdTypedef;

  struct SimpleStruct
  {
    string stringMember;
    long intMember;
    time typedefedMember;
    unsigned long uintMember;
    long long longMember;
    unsigned long long ulongMember;
    long longArr[5];
    IncludedIdl::IncludedStruct includedMember;
    ::IncludedIdl::IncludedStruct fullyScopedIncludedMember;
    string duplicate1, duplicate2;
  };

  typedef sequence <SimpleStruct> structs;

  typedef SimpleStruct typedefedStruct;

  struct StructWithArrays
  {
    structs typedefedArray;
    sequence<SimpleStruct> directArray;
  };

  enum SimpleEnum
  {
    EN_VAL_A,
    EN_VAL_B,
    EN_VAL_C
  };
  union SimpleUnion switch(SimpleEnum)
  {
    case EN_VAL_A:
      string strInUnion;
    case EN_VAL_B:
      unsigned long long ullInUnion;
    case EN_VAL_C:
      SimpleStruct simpleStructInUnion;
  };
  union WeirdUnion switch(long)
  {
    case 1+3:
    case 1 + 2 * 5:
    case 2 + 6 * (1 - 3) * ( -5/2) - 5:
      string foo;
    default:
      string bar;
  };

  exception NoMemberException {};

  exception ExceptionWithMembers
  {
    string someStringMember;
  };

  interface SimpleIdlInterface
  {
    string someStringMethod();
    SimpleEnum someMethodWithArgs(in string strArg, in unsigned long long ullArg, out SimpleStruct structArg);
    SimpleStruct someMethodWithArgsThatThrows(in string strArg, in unsigned long long ullArg, inout SimpleStruct structArg) raises(ExceptionWithMembers);
  };
};
