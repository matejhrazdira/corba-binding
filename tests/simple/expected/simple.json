{
  "fileName": "simple.idl",
  "definitions": [
    {
      "name": "IncludedIdl",
      "definitions": [
        {
          "name": "IncludedStruct",
          "members": [
            {
              "declarator": {
                "name": "someMember",
                "CLASS": "Declarator"
              },
              "type": {
                "CLASS": "StringType"
              },
              "CLASS": "Member"
            }
          ],
          "CLASS": "StructType"
        }
      ],
      "CLASS": "Module"
    },
    {
      "name": "SimpleIdl",
      "definitions": [
        {
          "name": "IDL_VERSION",
          "type": {
            "type": "SIGNED_LONG_INT"
          },
          "value": {
            "elements": [
              {
                "value": "1",
                "CLASS": "Literal"
              }
            ],
            "CLASS": "ConstExp"
          },
          "CLASS": "ConstDcl"
        },
        {
          "name": "COMPOSED_CONST",
          "type": {
            "type": "SIGNED_LONG_INT"
          },
          "value": {
            "elements": [
              {
                "value": "2",
                "CLASS": "Literal"
              },
              {
                "type": "ADD",
                "CLASS": "PrimitiveElement"
              },
              {
                "value": "6",
                "CLASS": "Literal"
              },
              {
                "type": "MULTIPLY",
                "CLASS": "PrimitiveElement"
              },
              {
                "type": "OPENING_BRACKET",
                "CLASS": "PrimitiveElement"
              },
              {
                "value": "1",
                "CLASS": "Literal"
              },
              {
                "type": "SUBTRACT",
                "CLASS": "PrimitiveElement"
              },
              {
                "value": "3",
                "CLASS": "Literal"
              },
              {
                "type": "CLOSING_BRACKET",
                "CLASS": "PrimitiveElement"
              },
              {
                "type": "MULTIPLY",
                "CLASS": "PrimitiveElement"
              },
              {
                "type": "OPENING_BRACKET",
                "CLASS": "PrimitiveElement"
              },
              {
                "type": "UNARY_MINUS",
                "CLASS": "PrimitiveElement"
              },
              {
                "value": "5",
                "CLASS": "Literal"
              },
              {
                "type": "DIVIDE",
                "CLASS": "PrimitiveElement"
              },
              {
                "value": "2",
                "CLASS": "Literal"
              },
              {
                "type": "CLOSING_BRACKET",
                "CLASS": "PrimitiveElement"
              },
              {
                "type": "SUBTRACT",
                "CLASS": "PrimitiveElement"
              },
              {
                "value": "5",
                "CLASS": "Literal"
              },
              {
                "type": "ADD",
                "CLASS": "PrimitiveElement"
              },
              {
                "name": [
                  "SimpleIdl",
                  "IDL_VERSION"
                ],
                "fullyQualified": true,
                "CLASS": "ScopedName"
              }
            ],
            "CLASS": "ConstExp"
          },
          "CLASS": "ConstDcl"
        },
        {
          "name": "STRING_CONSTANT",
          "type": {
            "CLASS": "StringType"
          },
          "value": {
            "elements": [
              {
                "value": "\"This is some string!\"",
                "CLASS": "Literal"
              }
            ],
            "CLASS": "ConstExp"
          },
          "CLASS": "ConstDcl"
        },
        {
          "name": "HEX_A",
          "type": {
            "type": "SIGNED_LONG_INT"
          },
          "value": {
            "elements": [
              {
                "value": "0x00001",
                "CLASS": "Literal"
              }
            ],
            "CLASS": "ConstExp"
          },
          "CLASS": "ConstDcl"
        },
        {
          "name": "HEX_B",
          "type": {
            "type": "SIGNED_LONG_INT"
          },
          "value": {
            "elements": [
              {
                "value": "0x00002",
                "CLASS": "Literal"
              }
            ],
            "CLASS": "ConstExp"
          },
          "CLASS": "ConstDcl"
        },
        {
          "name": "HEX_C",
          "type": {
            "type": "SIGNED_LONG_INT"
          },
          "value": {
            "elements": [
              {
                "name": [
                  "HEX_A"
                ],
                "fullyQualified": false,
                "CLASS": "ScopedName"
              },
              {
                "type": "OR",
                "CLASS": "PrimitiveElement"
              },
              {
                "name": [
                  "HEX_B"
                ],
                "fullyQualified": false,
                "CLASS": "ScopedName"
              }
            ],
            "CLASS": "ConstExp"
          },
          "CLASS": "ConstDcl"
        },
        {
          "type": {
            "type": "SIGNED_LONG_LONG_INT"
          },
          "declarators": [
            {
              "name": "time",
              "CLASS": "Declarator"
            }
          ],
          "CLASS": "TypeDeclarator"
        },
        {
          "type": {
            "elementType": {
              "name": [
                "time"
              ],
              "fullyQualified": false,
              "CLASS": "ScopedName"
            }
          },
          "declarators": [
            {
              "name": "TimeSequence",
              "CLASS": "Declarator"
            }
          ],
          "CLASS": "TypeDeclarator"
        },
        {
          "type": {
            "type": "SIGNED_LONG_LONG_INT"
          },
          "declarators": [
            {
              "name": "someReally",
              "CLASS": "Declarator"
            },
            {
              "name": "someReallyWeird",
              "CLASS": "Declarator"
            },
            {
              "name": "someReallyWeirdTypedef",
              "CLASS": "Declarator"
            }
          ],
          "CLASS": "TypeDeclarator"
        },
        {
          "name": "SimpleStruct",
          "members": [],
          "CLASS": "ForwardStructDcl"
        },
        {
          "name": "SimpleStruct",
          "members": [
            {
              "declarator": {
                "name": "stringMember",
                "CLASS": "Declarator"
              },
              "type": {
                "CLASS": "StringType"
              },
              "CLASS": "Member"
            },
            {
              "declarator": {
                "name": "intMember",
                "CLASS": "Declarator"
              },
              "type": {
                "type": "SIGNED_LONG_INT"
              },
              "CLASS": "Member"
            },
            {
              "declarator": {
                "name": "typedefedMember",
                "CLASS": "Declarator"
              },
              "type": {
                "name": [
                  "time"
                ],
                "fullyQualified": false,
                "CLASS": "ScopedName"
              },
              "CLASS": "Member"
            },
            {
              "declarator": {
                "name": "uintMember",
                "CLASS": "Declarator"
              },
              "type": {
                "type": "UNSIGNED_LONG_INT"
              },
              "CLASS": "Member"
            },
            {
              "declarator": {
                "name": "longMember",
                "CLASS": "Declarator"
              },
              "type": {
                "type": "SIGNED_LONG_LONG_INT"
              },
              "CLASS": "Member"
            },
            {
              "declarator": {
                "name": "ulongMember",
                "CLASS": "Declarator"
              },
              "type": {
                "type": "UNSIGNED_LONG_LONG_INT"
              },
              "CLASS": "Member"
            },
            {
              "declarator": {
                "dimensions": [
                  {
                    "elements": [
                      {
                        "value": "5",
                        "CLASS": "Literal"
                      }
                    ],
                    "CLASS": "ConstExp"
                  }
                ],
                "name": "longArr",
                "CLASS": "ArrayDeclarator"
              },
              "type": {
                "type": "SIGNED_LONG_INT"
              },
              "CLASS": "Member"
            },
            {
              "declarator": {
                "name": "includedMember",
                "CLASS": "Declarator"
              },
              "type": {
                "name": [
                  "IncludedIdl",
                  "IncludedStruct"
                ],
                "fullyQualified": false,
                "CLASS": "ScopedName"
              },
              "CLASS": "Member"
            },
            {
              "declarator": {
                "name": "fullyScopedIncludedMember",
                "CLASS": "Declarator"
              },
              "type": {
                "name": [
                  "IncludedIdl",
                  "IncludedStruct"
                ],
                "fullyQualified": true,
                "CLASS": "ScopedName"
              },
              "CLASS": "Member"
            },
            {
              "declarator": {
                "name": "duplicate1",
                "CLASS": "Declarator"
              },
              "type": {
                "CLASS": "StringType"
              },
              "CLASS": "Member"
            },
            {
              "declarator": {
                "name": "duplicate2",
                "CLASS": "Declarator"
              },
              "type": {
                "CLASS": "StringType"
              },
              "CLASS": "Member"
            }
          ],
          "CLASS": "StructType"
        },
        {
          "type": {
            "elementType": {
              "name": [
                "SimpleStruct"
              ],
              "fullyQualified": false,
              "CLASS": "ScopedName"
            }
          },
          "declarators": [
            {
              "name": "structs",
              "CLASS": "Declarator"
            }
          ],
          "CLASS": "TypeDeclarator"
        },
        {
          "type": {
            "name": [
              "SimpleStruct"
            ],
            "fullyQualified": false,
            "CLASS": "ScopedName"
          },
          "declarators": [
            {
              "name": "typedefedStruct",
              "CLASS": "Declarator"
            }
          ],
          "CLASS": "TypeDeclarator"
        },
        {
          "type": {
            "elementType": {
              "name": [
                "typedefedStruct"
              ],
              "fullyQualified": false,
              "CLASS": "ScopedName"
            }
          },
          "declarators": [
            {
              "name": "typedefedStructArray",
              "CLASS": "Declarator"
            }
          ],
          "CLASS": "TypeDeclarator"
        },
        {
          "name": "StructWithArrays",
          "members": [
            {
              "declarator": {
                "name": "typedefedArray",
                "CLASS": "Declarator"
              },
              "type": {
                "name": [
                  "structs"
                ],
                "fullyQualified": false,
                "CLASS": "ScopedName"
              },
              "CLASS": "Member"
            },
            {
              "declarator": {
                "name": "directArray",
                "CLASS": "Declarator"
              },
              "type": {
                "elementType": {
                  "name": [
                    "SimpleStruct"
                  ],
                  "fullyQualified": false,
                  "CLASS": "ScopedName"
                }
              },
              "CLASS": "Member"
            },
            {
              "declarator": {
                "name": "typedefTypedefedArray",
                "CLASS": "Declarator"
              },
              "type": {
                "name": [
                  "typedefedStructArray"
                ],
                "fullyQualified": false,
                "CLASS": "ScopedName"
              },
              "CLASS": "Member"
            }
          ],
          "CLASS": "StructType"
        },
        {
          "name": "SimpleEnum",
          "values": [
            {
              "name": "EN_VAL_A",
              "CLASS": "EnumValue"
            },
            {
              "name": "EN_VAL_B",
              "CLASS": "EnumValue"
            },
            {
              "name": "EN_VAL_C",
              "CLASS": "EnumValue"
            }
          ],
          "CLASS": "EnumType"
        },
        {
          "name": "SimpleUnion",
          "type": {
            "name": [
              "SimpleEnum"
            ],
            "fullyQualified": false,
            "CLASS": "ScopedName"
          },
          "fields": [
            {
              "type": {
                "CLASS": "StringType"
              },
              "declarator": {
                "name": "strInUnion",
                "CLASS": "Declarator"
              },
              "labels": [
                {
                  "expression": {
                    "elements": [
                      {
                        "name": [
                          "EN_VAL_A"
                        ],
                        "fullyQualified": false,
                        "CLASS": "ScopedName"
                      }
                    ],
                    "CLASS": "ConstExp"
                  },
                  "CLASS": "ExpressionLabel"
                }
              ],
              "CLASS": "UnionField"
            },
            {
              "type": {
                "type": "UNSIGNED_LONG_LONG_INT"
              },
              "declarator": {
                "name": "ullInUnion",
                "CLASS": "Declarator"
              },
              "labels": [
                {
                  "expression": {
                    "elements": [
                      {
                        "name": [
                          "EN_VAL_B"
                        ],
                        "fullyQualified": false,
                        "CLASS": "ScopedName"
                      }
                    ],
                    "CLASS": "ConstExp"
                  },
                  "CLASS": "ExpressionLabel"
                }
              ],
              "CLASS": "UnionField"
            },
            {
              "type": {
                "name": [
                  "SimpleStruct"
                ],
                "fullyQualified": false,
                "CLASS": "ScopedName"
              },
              "declarator": {
                "name": "simpleStructInUnion",
                "CLASS": "Declarator"
              },
              "labels": [
                {
                  "expression": {
                    "elements": [
                      {
                        "name": [
                          "EN_VAL_C"
                        ],
                        "fullyQualified": false,
                        "CLASS": "ScopedName"
                      }
                    ],
                    "CLASS": "ConstExp"
                  },
                  "CLASS": "ExpressionLabel"
                }
              ],
              "CLASS": "UnionField"
            }
          ],
          "CLASS": "UnionType"
        },
        {
          "name": "WeirdUnion",
          "type": {
            "type": "SIGNED_LONG_INT"
          },
          "fields": [
            {
              "type": {
                "CLASS": "StringType"
              },
              "declarator": {
                "name": "foo",
                "CLASS": "Declarator"
              },
              "labels": [
                {
                  "expression": {
                    "elements": [
                      {
                        "value": "1",
                        "CLASS": "Literal"
                      },
                      {
                        "type": "ADD",
                        "CLASS": "PrimitiveElement"
                      },
                      {
                        "value": "3",
                        "CLASS": "Literal"
                      }
                    ],
                    "CLASS": "ConstExp"
                  },
                  "CLASS": "ExpressionLabel"
                },
                {
                  "expression": {
                    "elements": [
                      {
                        "value": "1",
                        "CLASS": "Literal"
                      },
                      {
                        "type": "ADD",
                        "CLASS": "PrimitiveElement"
                      },
                      {
                        "value": "2",
                        "CLASS": "Literal"
                      },
                      {
                        "type": "MULTIPLY",
                        "CLASS": "PrimitiveElement"
                      },
                      {
                        "value": "5",
                        "CLASS": "Literal"
                      }
                    ],
                    "CLASS": "ConstExp"
                  },
                  "CLASS": "ExpressionLabel"
                },
                {
                  "expression": {
                    "elements": [
                      {
                        "value": "2",
                        "CLASS": "Literal"
                      },
                      {
                        "type": "ADD",
                        "CLASS": "PrimitiveElement"
                      },
                      {
                        "value": "6",
                        "CLASS": "Literal"
                      },
                      {
                        "type": "MULTIPLY",
                        "CLASS": "PrimitiveElement"
                      },
                      {
                        "type": "OPENING_BRACKET",
                        "CLASS": "PrimitiveElement"
                      },
                      {
                        "value": "1",
                        "CLASS": "Literal"
                      },
                      {
                        "type": "SUBTRACT",
                        "CLASS": "PrimitiveElement"
                      },
                      {
                        "value": "3",
                        "CLASS": "Literal"
                      },
                      {
                        "type": "CLOSING_BRACKET",
                        "CLASS": "PrimitiveElement"
                      },
                      {
                        "type": "MULTIPLY",
                        "CLASS": "PrimitiveElement"
                      },
                      {
                        "type": "OPENING_BRACKET",
                        "CLASS": "PrimitiveElement"
                      },
                      {
                        "type": "UNARY_MINUS",
                        "CLASS": "PrimitiveElement"
                      },
                      {
                        "value": "5",
                        "CLASS": "Literal"
                      },
                      {
                        "type": "DIVIDE",
                        "CLASS": "PrimitiveElement"
                      },
                      {
                        "value": "2",
                        "CLASS": "Literal"
                      },
                      {
                        "type": "CLOSING_BRACKET",
                        "CLASS": "PrimitiveElement"
                      },
                      {
                        "type": "SUBTRACT",
                        "CLASS": "PrimitiveElement"
                      },
                      {
                        "value": "5",
                        "CLASS": "Literal"
                      }
                    ],
                    "CLASS": "ConstExp"
                  },
                  "CLASS": "ExpressionLabel"
                }
              ],
              "CLASS": "UnionField"
            },
            {
              "type": {
                "CLASS": "StringType"
              },
              "declarator": {
                "name": "bar",
                "CLASS": "Declarator"
              },
              "labels": [
                {
                  "CLASS": "DefaultLabel"
                }
              ],
              "CLASS": "UnionField"
            }
          ],
          "CLASS": "UnionType"
        },
        {
          "name": "DefaultUnionFromEnum",
          "type": {
            "name": [
              "SimpleEnum"
            ],
            "fullyQualified": false,
            "CLASS": "ScopedName"
          },
          "fields": [
            {
              "type": {
                "CLASS": "StringType"
              },
              "declarator": {
                "name": "foo",
                "CLASS": "Declarator"
              },
              "labels": [
                {
                  "expression": {
                    "elements": [
                      {
                        "name": [
                          "EN_VAL_A"
                        ],
                        "fullyQualified": false,
                        "CLASS": "ScopedName"
                      }
                    ],
                    "CLASS": "ConstExp"
                  },
                  "CLASS": "ExpressionLabel"
                },
                {
                  "expression": {
                    "elements": [
                      {
                        "name": [
                          "EN_VAL_B"
                        ],
                        "fullyQualified": false,
                        "CLASS": "ScopedName"
                      }
                    ],
                    "CLASS": "ConstExp"
                  },
                  "CLASS": "ExpressionLabel"
                }
              ],
              "CLASS": "UnionField"
            },
            {
              "type": {
                "CLASS": "StringType"
              },
              "declarator": {
                "name": "bar",
                "CLASS": "Declarator"
              },
              "labels": [
                {
                  "CLASS": "DefaultLabel"
                }
              ],
              "CLASS": "UnionField"
            }
          ],
          "CLASS": "UnionType"
        },
        {
          "name": "NoMemberException",
          "members": [],
          "CLASS": "ExceptDcl"
        },
        {
          "name": "ExceptionWithMembers",
          "members": [
            {
              "declarator": {
                "name": "someStringMember",
                "CLASS": "Declarator"
              },
              "type": {
                "CLASS": "StringType"
              },
              "CLASS": "Member"
            }
          ],
          "CLASS": "ExceptDcl"
        },
        {
          "innerDefinitions": [
            {
              "name": "NESTED_CONST",
              "type": {
                "type": "SIGNED_LONG_INT"
              },
              "value": {
                "elements": [
                  {
                    "value": "123",
                    "CLASS": "Literal"
                  }
                ],
                "CLASS": "ConstExp"
              },
              "CLASS": "ConstDcl"
            },
            {
              "name": "NestedException",
              "members": [],
              "CLASS": "ExceptDcl"
            }
          ],
          "inheritance": [],
          "operations": [
            {
              "name": "someStringMethod",
              "oneway": false,
              "returnType": {
                "CLASS": "StringType"
              },
              "params": [],
              "exceptions": [],
              "CLASS": "Operation"
            },
            {
              "name": "methodWithTypedefedValues",
              "oneway": false,
              "returnType": {
                "name": [
                  "time"
                ],
                "fullyQualified": false,
                "CLASS": "ScopedName"
              },
              "params": [
                {
                  "name": "timeArg",
                  "direction": "IN",
                  "type": {
                    "name": [
                      "time"
                    ],
                    "fullyQualified": false,
                    "CLASS": "ScopedName"
                  },
                  "CLASS": "Param"
                },
                {
                  "name": "structArg",
                  "direction": "OUT",
                  "type": {
                    "name": [
                      "typedefedStruct"
                    ],
                    "fullyQualified": false,
                    "CLASS": "ScopedName"
                  },
                  "CLASS": "Param"
                }
              ],
              "exceptions": [],
              "CLASS": "Operation"
            },
            {
              "name": "someMethodWithArgs",
              "oneway": false,
              "returnType": {
                "name": [
                  "SimpleEnum"
                ],
                "fullyQualified": false,
                "CLASS": "ScopedName"
              },
              "params": [
                {
                  "name": "strArg",
                  "direction": "IN",
                  "type": {
                    "CLASS": "StringType"
                  },
                  "CLASS": "Param"
                },
                {
                  "name": "ullArg",
                  "direction": "IN",
                  "type": {
                    "type": "UNSIGNED_LONG_LONG_INT"
                  },
                  "CLASS": "Param"
                },
                {
                  "name": "structArg",
                  "direction": "OUT",
                  "type": {
                    "name": [
                      "SimpleStruct"
                    ],
                    "fullyQualified": false,
                    "CLASS": "ScopedName"
                  },
                  "CLASS": "Param"
                }
              ],
              "exceptions": [],
              "CLASS": "Operation"
            },
            {
              "name": "someMethodWithArgsThatThrows",
              "oneway": false,
              "returnType": {
                "name": [
                  "SimpleStruct"
                ],
                "fullyQualified": false,
                "CLASS": "ScopedName"
              },
              "params": [
                {
                  "name": "strArg",
                  "direction": "IN",
                  "type": {
                    "CLASS": "StringType"
                  },
                  "CLASS": "Param"
                },
                {
                  "name": "ullArg",
                  "direction": "IN",
                  "type": {
                    "type": "UNSIGNED_LONG_LONG_INT"
                  },
                  "CLASS": "Param"
                },
                {
                  "name": "structArg",
                  "direction": "IN_OUT",
                  "type": {
                    "name": [
                      "SimpleStruct"
                    ],
                    "fullyQualified": false,
                    "CLASS": "ScopedName"
                  },
                  "CLASS": "Param"
                }
              ],
              "exceptions": [
                {
                  "name": [
                    "ExceptionWithMembers"
                  ],
                  "fullyQualified": false,
                  "CLASS": "ScopedName"
                }
              ],
              "CLASS": "Operation"
            },
            {
              "name": "getTimeSequence",
              "oneway": false,
              "returnType": {
                "name": [
                  "TimeSequence"
                ],
                "fullyQualified": false,
                "CLASS": "ScopedName"
              },
              "params": [],
              "exceptions": [],
              "CLASS": "Operation"
            },
            {
              "name": "throwsNestedException",
              "oneway": false,
              "returnType": {},
              "params": [],
              "exceptions": [
                {
                  "name": [
                    "NestedException"
                  ],
                  "fullyQualified": false,
                  "CLASS": "ScopedName"
                }
              ],
              "CLASS": "Operation"
            }
          ],
          "name": "SimpleIdlInterface",
          "CLASS": "Declaration"
        },
        {
          "innerDefinitions": [],
          "inheritance": [
            {
              "name": [
                "SimpleIdlInterface"
              ],
              "fullyQualified": false,
              "CLASS": "ScopedName"
            }
          ],
          "operations": [
            {
              "name": "someAdditionalMethod",
              "oneway": false,
              "returnType": {
                "CLASS": "StringType"
              },
              "params": [],
              "exceptions": [],
              "CLASS": "Operation"
            }
          ],
          "name": "InheritedInterface",
          "CLASS": "Declaration"
        }
      ],
      "CLASS": "Module"
    }
  ],
  "CLASS": "Specification"
}